_posts/2017-03-05-artsy-technology-stack-2017.md

```diff
@@ -0,0 +1,169 @@
+
+<center>
+ <img src="/images/tech-2017/engineering-teams.svg" style="width:100%;">
+</center>
+
+# Artsy Technology Infrastructure
+
+<center>
+ <img src="/images/tech-2017/languages.svg" style="width:100%;">
+</center>
+
+
+## User Facing
+
+A lot of the user-facing focus includes being able to present interfaces with a quality worthy of art.
+
+What you see today when you go to [www.artsy.net](https://artsy.net) is a website built with [Ezel.js](http://ezeljs.com), which is a boilerplate for [Backbone](http://backbonejs.org) projects running on [Node](https://nodejs.org) and using [Express](http://expressjs.com) and [Browserify](http://browserify.org). We used to have separate projects for mobile and desktop web, but they [are now merged](https://github.com/artsy/force/pull/890). The combined app is hosted on [Heroku](http://heroku.com) and uses [Redis](http://redis.io) for caching. Assets, including artwork images, are served from [Amazon S3](http://aws.amazon.com/s3) via the [CloudFront CDN](http://aws.amazon.com/cloudfront). This [code is open-source](https://github.com/artsy/force).
+
+What you see today when you open the [Artsy iOS app](https://itunes.apple.com/us/app/artsy-collect-and-bid-on-fine-art-design/id703796080?mt=8) is a mix of Objective-C, Swift and React Native. Objective-C and Swift continue to provide a lot of over-arching cross-View Controller code. While individual representations of Artsy resources tend to be built in React Native. All of our React Native code uses Relay to handle API integration. This [code is open-source](https://github.com/artsy/eigen).
+
+You can also find Artsy on [Alexa](http://alexa.atsy.net) and [Google Home](http://assistant.artsy.net), which are both open-source Node.js applications.
+
+Our core API serves the public facets of our product, many of our own internal applications, and even [some of your own projects](https://developers.artsy.net). It's built with [Ruby](https://www.ruby-lang.org/en), [Rack](http://rack.github.io), [Rails](http://rubyonrails.org), and [Grape](https://github.com/intridea/grape) serving primarily JSON. The API is hosted on [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Memcached](http://memcached.org) for caching and [Redis](https://redis.io) for background queues. It runs background jobs with [delayed_job](https://github.com/collectiveidea/delayed_job). We used to employ [Apache Solr](http://lucene.apache.org/solr) and even [Google Custom Search](https://www.google.com/cse) for the many search functions, but have since consolidated on [Elasticsearch](https://www.elastic.co).
+
```

Comment:
Cool, I've touched on this, think its a nice distinction to be able to mention developers.artsy.net also - thanks